.github/workflows/formatting.yml

```diff
@@ -0,0 +1,103 @@
+name: formatting-check
```

Comment:
@catenacyber Here's the github integration.

Reply:
Thanks, does this GitHub action take care of the Rust code as well ? (with the already included rustfmt.toml file)

Reply:
Hah! No. I did not touch that elephant in the room with a ten-foot pole!

Yeah, indeed rustfmt would be good to run as not all rust code is properly formatted I've been told. As I am a hard-core C++ programmer that dabbles in C when required, but not rust, I decided not to touch this. 

Anyhow, similar approach should be chosen, i.e. only changes should be formatted, I guess. Not sure if rustfmt supports just formatting changes similar to git-clang-format. If not, we could probably write our own git-clang-format for rust. git-rustfmt anyone? 

Let's make this a separate task.

Reply:
Unfortunately I don't think rustfmt is advanced enough in this area.  From what I can tell its the complete file or nothing, and being able to just look at the changed lines is still an open issue:
- https://github.com/rust-lang/rustfmt/issues/4103
- https://github.com/rust-lang/rustfmt/issues/1324

IMO for the purposes of this PR, keep it to the C only.

Reply:
FYI, opened https://redmine.openinfosecfoundation.org/issues/3836 to track rust formatting